### PR TITLE
Razor Synthax

### PR DESCRIPTION
**Pull request by pmdevers**
Added Razor Synthax and common used keywords

### CODE REVIEW

File: lib/ace/ext/modelist.js [hunk 1]
```diff
@@ -167,7 +167,8 @@ var supportedModes = {
     XQuery:      ["xq"],
     YAML:        ["yaml|yml"],
     // Add the missing mode "Django" to ext-modelist
-    Django:      ["html"]
+    Django:      ["html"],
+	Razor:		 ["razor"]
```
Comment (nightwing) [inline lib/ace/ext/modelist.js]: this is supposed to be a list of extensions for the mode, cshtml and maybe some other extensions.

**Comment by nightwing** [comment]
To merge this i have to ask you to submit a CLA see [CONTRIBUTING.md](https://github.com/ajaxorg/ace/blob/master/CONTRIBUTING.md) for details.
How much of this is based on https://github.com/ashmind/ace/tree/razor-wip-shelf? We may have to ask a CLA from it's author too.

**Comment by nightwing** [comment]
Cherry picked and merged as https://github.com/ajaxorg/ace/commit/cb6dd5514da2b5dda41d10b117ff26e515e711b6, thanks!

**Comment by iJungleboy** [comment]
It appears to not really work - at least the `@` isn't highlighted (typically yellow), and normal HTML tags don't auto-complete any more.

**Pull request closed without merging**